smart-contracts/contracts/mixins/MixinPurchase.sol

```diff
@@ -83,13 +88,20 @@ contract MixinPurchase is
         idTo
       );
     } else if (toKey.expirationTimestamp > block.timestamp) {
+      // prevent re-purchase of a valid non-expiring key
+      require(expirationDuration != type(uint).max, 'A valid non-expiring key can not be purchased twice');
```

Comment:
actually now that I think about it I think we should block the double purchase only if the existing key is set to expire in maxuint.
So basically we should allow double purchase if/when the existing key expires in the future (or has expired) but is less than maxuint. This would cover cases where the duration has been changed to maxuint. wdyt?

Reply:
yes good